proving_system/integration_cairo/src/main.rs

```diff
@@ -0,0 +1,219 @@
+    let memory = CairoMemory::from_file(&dir_memory).unwrap();
+
+    let execution_trace = build_cairo_execution_trace(&raw_trace, &memory);
+
+    // This trace is obtained from Giza when running the prover for the mentioned program.
+    let expected_trace = TraceTable::new_from_cols(&vec![
+        // col 0
+        vec![FE::zero(), FE::zero(), FE::one()],
+        // col 1
+        vec![FE::one(), FE::one(), FE::one()],
+        // col 2
+        vec![FE::one(), FE::one(), FE::zero()],
+        // col 3
+        vec![FE::zero(), FE::zero(), FE::one()],
+        // col 4
+        vec![FE::zero(), FE::zero(), FE::zero()],
+        // col 5
+        vec![FE::zero(), FE::zero(), FE::zero()],
+        // col 6
+        vec![FE::zero(), FE::zero(), FE::zero()],
+        // col 7
+        vec![FE::zero(), FE::zero(), FE::one()],
+        // col 8
+        vec![FE::zero(), FE::zero(), FE::zero()],
+        // col 9
+        vec![FE::zero(), FE::zero(), FE::zero()],
+        // col 10
+        vec![FE::zero(), FE::zero(), FE::zero()],
+        // col 11
+        vec![FE::one(), FE::zero(), FE::zero()],
+        // col 12
+        vec![FE::zero(), FE::zero(), FE::zero()],
+        // col 13
+        vec![FE::zero(), FE::zero(), FE::one()],
+        // col 14
+        vec![FE::one(), FE::one(), FE::zero()],
+        // col 15
+        vec![FE::zero(), FE::zero(), FE::zero()],
+        // col 16
+        vec![FE::from(3), FE::from(3), FE::from(9)],
+        // col 17
+        vec![FE::from(8), FE::from(9), FE::from(9)],
+        // col 18
+        vec![FE::from(8), FE::from(8), FE::from(8)],
+        // col 19
+        vec![FE::from(1), FE::from(3), FE::from(5)],
+        // col 20
+        vec![FE::from(8), FE::from(8), FE::from(6)],
+        // col 21
+        vec![FE::from(7), FE::from(7), FE::from(7)],
+        // col 22
+        vec![FE::from(2), FE::from(4), FE::from(7)],
+        // col 23
+        vec![
+            FE::from(0x480680017fff8000),
+            FE::from(0x400680017fff7fff),
+            FE::from(0x208b7fff7fff7ffe),
+        ],
+        // col 24
+        vec![FE::from(3), FE::from(3), FE::from(9)],
+        // col 25
+        vec![FE::from(9), FE::from(9), FE::from(9)],
+        // col 26
+        vec![FE::from(3), FE::from(3), FE::from(9)],
+        // col 27
+        vec![FE::from(0x8000), FE::from(0x7fff), FE::from(0x7ffe)],
+        // col 28
+        vec![FE::from(0x7fff), FE::from(0x7fff), FE::from(0x7fff)],
+        // col 29
+        vec![FE::from(0x8001), FE::from(0x8001), FE::from(0x7fff)],
+        // col 30
+        vec![FE::zero(), FE::zero(), FE::zero()],
+        // col 31
+        vec![FE::zero(), FE::zero(), FE::zero()],
+        // col 32
+        vec![FE::from(0x1b), FE::from(0x1b), FE::from(0x51)],
+        // col 33 - Selector column
+        // vec![FE::one(), FE::one(), FE::zero()],
+    ]);
+
+    assert_eq!(execution_trace.cols(), expected_trace.cols());
+}
+
+fn run(
+    entrypoint: &str,
```

Comment:
This argument will be almost always the "main" string, I think we could make it an option and set "main" as the default value when it is None